src/ui/pages/SendTransaction/SendTransaction.tsx

```diff
@@ -370,166 +587,190 @@ function TransactionDefaultView({
+    transaction: populatedTransaction,
+    address: singleAddress,
+    origin,
+    enabled: FEATURE_PAYMASTER_ENABLED
+      ? eligibility?.data.eligible === false
+      : true,
+  });
+
+  const paymasterTxInterpretQuery = useQuery({
+    enabled: paymasterEligible,
+    suspense: false,
     queryKey: [
-      'interpretTransaction',
-      incomingTxWithGasAndFee,
+      'interpret/typedData',
+      populatedTransaction,
+      chainGasPrices,
+      configuration,
+      allowanceQuantityBase,
+      chain,
       singleAddress,
-      origin,
+      networks,
+      transactionAction,
     ],
-    queryFn: () => {
-      return incomingTxWithGasAndFee
-        ? interpretTransaction({
-            address: singleAddress,
-            transaction: incomingTxWithGasAndFee,
-            origin,
-          })
-        : null;
+    queryFn: async () => {
+      const configuredTx = await configureTransactionToSign(
+        populatedTransaction,
+        {
+          chainGasPrices: chainGasPrices || null,
+          configuration,
+          allowanceQuantityBase,
+          chain,
+          from: singleAddress,
+          networks,
+          transactionAction,
+        }
+      );
+      const toSign = await assignPaymaster(configuredTx);
+      const typedData = await walletPort.request('uiGetEip712Transaction', {
+        transaction: toSign,
+      });
+      return interpretSignature({
+        address: toSign.from,
+        chainId: normalizeChainId(toSign.chainId),
+        typedData,
+      });
     },
-    enabled: Boolean(incomingTxWithGasAndFee),
-    keepPreviousData: true,
-    staleTime: 20000,
-    suspense: false,
-    retry: 1,
   });
 
+  const interpretQuery = paymasterEligible
+    ? paymasterTxInterpretQuery
+    : txInterpretQuery;
+
+  const interpretationHasCriticalWarning = hasCriticalWarning(
+    interpretQuery.data?.warnings
+  );
+
   const requestedAllowanceQuantityBase =
-    interpretation?.action?.content?.single_asset?.quantity ||
+    interpretQuery.data?.action?.content?.single_asset?.quantity ||
     localAddressAction?.content?.single_asset?.quantity;
+  const interpretAddressAction = interpretQuery.data?.action;
 
-  const [allowanceQuantityBase, setAllowanceQuantityBase] = useState('');
-
-  const interpretAddressAction = interpretation?.action;
+  const addressAction = interpretAddressAction || localAddressAction || null;
 
   const view = params.get('view') || View.default;
   const advancedDialogRef = useRef<HTMLDialogElementInterface | null>(null);
   const openAdvancedView = useCallback(() => {
     advancedDialogRef.current?.showModal();
   }, []);
 
+  const feeValueCommonRef = useRef<string>(); /** for analytics only */
+  const handleFeeValueCommonReady = useCallback((value: string) => {
+    feeValueCommonRef.current = value;
+  }, []);
+
+  const next = params.get('next');
+  function handleSentTransaction(tx: ethers.providers.TransactionResponse) {
```

Comment:
Should it be inside useCallback?

Reply:
We're not passing it to any of the components, so currently would not matter